keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/UnhealthyVetoRepository.kt

```diff
@@ -0,0 +1,58 @@
+package com.netflix.spinnaker.keel.persistence
+
+import org.slf4j.LoggerFactory
+import java.time.Clock
+import java.time.Instant
+
+abstract class UnhealthyVetoRepository(
+  open val clock: Clock
+) {
+  val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  init {
+    log.info("Using ${javaClass.simpleName}")
+  }
+
+  /**
+   * @return true if we don't have an unhealthy record, or if a resource has
+   * been marked as unhealthy only once.
+   *
+   * We want to see a bit of sustained unhealthy duration before we
+   * take any action.
+   */
+  abstract fun isHealthy(
+    resourceId: String
+  ): Boolean
+
+  abstract fun markUnhealthy(
+    resourceId: String,
+    application: String
+  )
+
+  abstract fun markAllowed(
```

Comment:
If the resource is unhealthy, we're still going to check it every 5 minutes. "Allowed time" (words are hard) represents the time we last allowed the resource to be checked instead of vetoing it. Marking it healthy isn't correct because the resource is still unhealthy